---
title: "An International Landscape of Cancer NGS Reporting Practices"
output: 
  flexdashboard::flex_dashboard:
    theme:
      bg: "#fdfdfd"
      fg: "#18252b" 
      primary: "#18252b"
      base_font:
        google: Prompt
      code_font:
        google: JetBrains Mono
      base_size: 24
    orientation: columns
    vertical_layout: fill
    fig_mobile: TRUE
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(flexdashboard)
library(tidyverse)
library(flextable)
library(ggpubr)
library(ggpmisc)
library(ggrepel)
library(DT)
library(ComplexUpset)
library(sf)
library(cowplot)
library(grid)
library(ggplotify)
# Install thematic and un-comment for themed static plots (i.e., ggplot2)
# thematic::thematic_rmd()
```

```{css, echo=FALSE}
.fluid-row {
  font-size: 5.9vw;
}
```

```{r load-data, include=FALSE}
data <- read_csv("data/NGSreportElementsClean.csv")
data.essentiality <- read_csv("data/NGSreportEssentiality.csv")
data.essentiality.europe <- read_csv("data/NGSreportEssentialityEurope.csv")
data.reported <- read_csv("data/NGSreportElementsReported.csv")
data.reported.europe <- read_csv("data/NGSreportElementsReportedEurope.csv")
```

<!---
```{r doubleResponders, include=FALSE}
data |> 
  select(timestamp, ID) |> 
  distinct() |> 
  group_by(ID) |> 
  filter(n()>1) |> 
  select(ID) |> 
  distinct() |> 
  nrow() -> nrDoubleResponders
```


```{r selectResponders}
data |> 
  group_by(ID) |> 
  slice_max(timestamp) |> 
  ungroup() -> data
```
--->

```{r nrResponders}
data |> 
  select(continent, ID) |> 
  distinct() |> 
  count(continent) |> 
  pivot_wider(names_from = "continent", values_from = "n") -> nrResp
```


```{r colorScheme}
palContinents <- c(Europe = "#283F46", `North America` = "#3aada8", Other = "gray", Worldwide = "#8B0000")

textSize <- 24
cutoff <- 0.75
```

Results
=====================================  

Results of Next Generation Sequencing (NGS) are increasingly incorporated in routine clinical oncology. Multiple guidelines and standards have been issued globally to support clinical NGS variant interpretation and are currently integrated in different ways into cancer NGS reports around the world. However, a thorough capture of how these guidelines are being translated in real-world laboratory  reporting  has not yet been clearly studied. To this end, the Variant Interpretation for Cancer Consortium Virtual Molecular Tumor Board (VICC-VMTB), in collaboration with the Cancer Genomics Consortium (CGC) and the ClinGen Somatic Clinical Domain Working Group (CDWG), distributed a comprehensive survey regarding reporting practices for cancer NGS testing worldwide to identify essential elements and their actual occurrence in NGS reports.



Column {data-width=350}
-----------------------------------------------------------------------

### Global representation of survey responses

```{r participants, fig.width=20, fig.height=10, fig.cap=paste("Overview of survey responses across the globe. Most respondents represented laboratories in North America (n = ", nrResp$`North America`,") and Europe (n = ", nrResp$Europe, "), with additional responses from other countries (n = ", nrResp$Other, "). ")}
data |> 
  select(region, ID, continent) |> 
  distinct() |> 
  mutate(region = ifelse(region == "UK", "United Kingdom", region)) |> 
  mutate(region = ifelse(region == "USA", "United States of America", region)) |> 
  group_by(region) |> 
  count() -> sum.country

sf::sf_use_s2(FALSE)
world <- rnaturalearth::ne_countries(scale = "small", returnclass = "sf")
world |> 
  cbind(st_coordinates(st_centroid(world$geometry))) |> 
  mutate(region = name) |> 
  left_join(sum.country) |> 
  ggplot() + geom_sf(aes(fill = n)) +
  scale_fill_continuous(low="thistle2", high="darkred", 
                        guide="colorbar",na.value="lightgray", 
                        name="# Responses",
                        breaks=seq(0, 20, by = 5)) +
  geom_label_repel(data = world |> 
                     cbind(st_coordinates(st_centroid(world$geometry))) |> 
                     mutate(region = name) |> 
                     left_join(sum.country) |> filter(n > 1) ,aes(X, Y, label = n), 
                   size = textSize/2, nudge_x = -5, nudge_y = -1) +
  labs(x="", y="") + 
  theme_pubr(base_size = textSize, legend = "bottom") -> plot.countries
  

# add bar-plot with country not given
sum.country |> 
  filter(is.na(region)) |> 
  mutate(region = str_replace_na("Country not specified")) |> 
  pivot_wider(names_from = "region", values_from = "n")-> na.region

plot.countries +
  annotate("table", x = -180, y = -70, label = list(cbind(na.region))) +
  ggpubr::theme_transparent() +
  theme(base_size = textSize, legend.position = "bottom") -> plot.countries

plot.countries
#ggsave("figures/overviewParticipants.png", width = 14, height = 10.5, dpi = 600)
```

### Background of participants

```{r institutes, fig.width=20, fig.height=10, fig.cap=paste("Histogram of participant work settings at a clinical, commercial, and/or academic setting.")}
org.factor <- c("Clinical/Hospital Laboratory", "Commercial Laboratory", "Academic/Research Laboratory", "Other")
data |> 
  filter(name == "organization") |> 
  select(ID, name, value) |> 
  distinct() |> 
  group_by(ID) |> 
  mutate(org = 1:n()) |> 
  mutate(org = paste0("org", org, sep="")) |> 
  pivot_wider(names_from = org, values_from = value) |> 
  select(-name) |> 
  mutate(org1 = ifelse(org1 %in% org.factor, org1, "Other")) |> 
  group_by(org1, org2) |> 
  count() |> 
  mutate(org2 = ifelse(is.na(org2), org1, org2)) |>  
  mutate(org1 = factor(org1, levels = org.factor),
         org2 = factor(org2, levels = org.factor))  -> sum.organization

data |> 
  filter(name == "organization") |> 
  select(ID, name, value) |> 
  distinct() |> 
  pivot_wider(names_from = name, values_from = value) |> 
  unnest(organization) |> 
  mutate(organization = ifelse(organization %in% org.factor, organization, "Other")) |> 
  mutate(count = 1) |> 
  pivot_wider(names_from = organization, values_from = count, values_fill = 0) -> data.upset
data.upset[org.factor] <- data.upset[org.factor] == 1

data.upset |> 
  upset(intersect = org.factor, name = "", 
        set_sizes = (upset_set_size() + ylab("# Responses")),
        base_annotations = list(
          'Intersection size' = (intersection_size() + ylab("# Responses")) 
          )) +
  theme_pubr(base_size = textSize) -> plot.participants 
plot.participants

#ggsave("figures/overviewBackground.png", width = 14, height = 7, dpi = 600)
```


### Types of tumors analysed
```{r plotTumortypes, fig.width=20, fig.height=10}
cancer.factor <- rev(c("Solid Tumor", "Hematologic Malignancies", 
                       "Both", "Both (only pediatric cancer)", "Other", "Not specified"))
data |> 
  filter(name == "cancer_type") |> 
  select(ID, value) |> 
  distinct() |> 
  group_by(value) |> 
  replace_na(list(value = "Not specified")) |> 
  mutate(value = ifelse(value %in% cancer.factor, value, "Other")) |> 
  count() |> 
  mutate(cancer_type = factor(value, levels = cancer.factor)) |> 
  filter(cancer_type != "Not specified") |> 
  ggbarplot(x = "cancer_type", y = "n", fill = "grey23", 
            ylab = "# Responses", xlab = "", rotate = T) +
  #scale_fill_continuous(low="thistle2", high="darkred") +
  theme_pubr(base_size = textSize) +
  theme(legend.position = "none") -> plot.cancer
plot.cancer
#ggsave("figures/overviewTumors.png", width = 14, height = 7, dpi = 600)
```


Column {data-width=650 .tabset}
-----------------------------------------------------------------------

### General Information on NGS analysis
```{r plotOverview, fig.width=40, fig.height=20, fig.cap=("Participants were asked whether each of the data elements above was considered essential on NGS reports. A score of essentiality was calculated from these responses and is displayed by region of respondent.")}
data.essentiality |> 
  filter(section == "Overview", continent == "Worldwide") |> 
  arrange(normEssent) |> 
  mutate(label = paste0(name, ", SE = ", round(normEssent, 2))) -> orderOverview

data.essentiality |> 
  filter(section == "Overview") |> 
  mutate(name = factor(name, levels = orderOverview$name)) |> 
  ggbarplot(x = "name", y = "normEssent", palette = palContinents, 
            fill = "continent", 
            rotate = T, 
            label = TRUE, lab.size = textSize/3, lab.nb.digits = 2, lab.pos = "in", lab.col = "white", 
            lab.vjust = 0.5, lab.hjust = 2,
            xlab = "", ylab = "Score of Essentiality", position = position_dodge(0.9)) + 
#  ylim(-0.16, 1)+
  geom_hline(yintercept = cutoff, color = "#8B0000", lty = "dashed", size = 2) +
  theme_pubr(base_size = textSize*2) + 
  theme(legend.position = "top", legend.title=element_blank()) 
```

### Overview of variant reporting
```{r plotVariants, fig.width=40, fig.height=20, fig.cap=("Participants were asked whether each of the data elements above was considered essential on NGS reports. A score of essentiality was calculated from these responses and is displayed by region of respondent.")}
data.essentiality |> 
  filter(section == "Variants") |> 
  filter(continent == "Worldwide") |> 
  arrange(normEssent) |> 
  mutate(label = paste0(name, ", SE = ", round(normEssent, 2))) -> orderVariants

data.essentiality |> 
  filter(section == "Variants") |> 
  mutate(name = factor(name, levels = orderVariants$name)) |> 
  ggbarplot(x = "name", y = "normEssent", palette = palContinents, 
            fill = "continent", 
            rotate = T, 
            label = TRUE, lab.size = textSize/3, lab.nb.digits = 2, lab.pos = "in", lab.col = "white", 
            lab.vjust = 0.5, lab.hjust = 2,
            xlab = "", ylab = "Score of Essentiality", position = position_dodge(0.9)) + 
#  ylim(-0.16, 1)+
  geom_hline(yintercept = cutoff, color = "#8B0000", lty = "dashed", size = 2) +
  theme_pubr(base_size = textSize*2) + 
  theme(legend.position = "top", legend.title=element_blank()) 
```

### Overview of functional assessment
```{r plotFunctional, fig.width=40, fig.height=20, fig.cap=("Participants were asked whether each of the data elements above was considered essential on NGS reports. A score of essentiality was calculated from these responses and is displayed by region of respondent.")}
data.essentiality |> 
  filter(section == "Functional") |> 
  filter(continent == "Worldwide") |> 
  arrange(normEssent) |> 
  mutate(label = paste0(name, ", SE = ", round(normEssent, 2))) -> orderFunctional


data.essentiality |> 
  filter(section == "Functional") |> 
  mutate(name = factor(name, levels = orderFunctional$name)) |> 
  ggbarplot(x = "name", y = "normEssent", palette = palContinents, 
            fill = "continent", 
            rotate = T, 
            label = TRUE, lab.size = textSize/3, lab.nb.digits = 2, lab.pos = "in", lab.col = "white", 
            lab.vjust = 0.5, lab.hjust = 2,
            xlab = "", ylab = "Score of Essentiality", position = position_dodge(0.9)) + 
  ylim(-0.16, 1)+
  geom_hline(yintercept = cutoff, color = "#8B0000", lty = "dashed", size = 2) +
  theme_pubr(base_size = textSize*2) + 
  theme(legend.position = "top", legend.title=element_blank()) 
```

### Treatment recommendations
```{r plotTreatments, fig.width=40, fig.height=2, fig.cap=("Participants were asked whether each of the data elements above was considered essential on NGS reports. A score of essentiality was calculated from these responses and is displayed by region of respondent.")}
data.essentiality |> 
  filter(section == "Treatment") |> 
  filter(continent == "Worldwide") |> 
  arrange(normEssent) |> 
  mutate(label = paste0(name, ", SE = ", round(normEssent, 2))) -> orderTreatment


data.essentiality |> 
  filter(section == "Treatment") |> 
  mutate(name = factor(name, levels = orderTreatment$name)) |> 
  ggbarplot(x = "name", y = "normEssent", palette = palContinents, 
            fill = "continent", 
            rotate = T, 
            label = TRUE, lab.size = textSize/3, lab.nb.digits = 2, lab.pos = "in", lab.col = "white", 
            lab.vjust = 0.5, lab.hjust = 2,
            xlab = "", ylab = "Score of Essentiality", position = position_dodge(0.9)) + 
#  ylim(-0.16, 1)+
  geom_hline(yintercept = cutoff, color = "#8B0000", lty = "dashed", size = 2) +
  theme_pubr(base_size = textSize*2) +
  theme(legend.position = "top", legend.title=element_blank()) 
```

### Clinical trials
```{r plotClinicalTrials, fig.width=40, fig.height=20, fig.cap=("Participants were asked whether each of the data elements above was considered essential on NGS reports. A score of essentiality was calculated from these responses and is displayed by region of respondent.")}
data.essentiality |> 
  filter(group == "clinical trial") |> 
  filter(continent == "Worldwide") |> 
  arrange(normEssent) |> 
  pull(name) -> listPlot


data.essentiality |> 
  filter(group == "clinical trial") |> 
  mutate(name = factor(name, levels = listPlot)) |> 
  ggbarplot(x = "name", y = "normEssent", palette = palContinents, 
            fill = "continent", 
            rotate = T, 
            label = TRUE, lab.size = textSize/3, lab.nb.digits = 2, lab.pos = "in", lab.col = "white", 
            lab.vjust = 0.5, lab.hjust = 2,
            xlab = "", ylab = "Score of Essentiality", position = position_dodge(0.9)) + 
#  ylim(-0.16, 1)+
  geom_hline(yintercept = cutoff, color = "#8B0000", lty = "dashed", size = 2) +
  theme_pubr(base_size = textSize*2) + 
  theme(legend.position = "top", legend.title=element_blank())
```

Reporting
====================================
Row
-------------------------------------
```{r calcReported}
responses.reported <- c("No Response", "Not Reported", "Reported")
data |> 
  filter(class == "Report") |> 
  group_by(continent, name, group, section) |> 
  count(value) |> 
  pivot_wider(names_from = value, values_from = n) |> 
  left_join(nrResp |> pivot_longer(everything(), names_to = "continent")) |> 
  mutate(sumResponses = sum(Reported, `Not Reported`, na.rm = T),
         `No Response` = value - sumResponses) |> 
  mutate(percReported = Reported / sumResponses) -> data.reported

data |> 
  filter(class == "Report") |> 
  group_by(name, group, section) |> 
  count(value) |> 
  pivot_wider(names_from = value, values_from = n) |> 
  mutate(value = nrResp |> as_vector() |> sum()) |> 
  mutate(sumResponses = sum(Reported, `Not Reported`, na.rm = T),
         `No Response` = value - sumResponses) |> 
  mutate(percReported = Reported / sumResponses) |> 
  mutate(continent = "Worldwide") |> 
  bind_rows(data.reported) -> data.reported

```
    
### General Information on NGS analysis
    
```{r reportOverview, fig.width=20, fig.height=10}
data.essentiality |> 
  filter(section == "Overview") |> 
  left_join(data.reported, by = c("group", "section", "name", "continent")) |> 
  mutate(label = ifelse((continent == "Worldwide" & normEssent >= cutoff), name, "")) |> 
  ggscatter(x = "normEssent", y = "percReported", palette = palContinents, 
            color = "continent", size = 4,
            xlab = "Score of Essentiality (SE)", ylab = "% Reported") + 
  lims(x = c(0, 1), y = c(0,1))+
  geom_abline(slope = 1, intercept = 0, lty = "dashed", size = 1, color = "grey23")+
  geom_label_repel(aes(label = label, color = continent), box.padding = 1, max.overlaps = Inf, 
                   show.legend = F, size = textSize/4)+
  theme_pubclean(base_size = textSize/2) + 
  theme(legend.position = "none") -> p1

data.reported |> 
  filter(section == "Overview") |> 
  pivot_longer(cols = responses.reported, values_to = "count", names_to = "report") |> 
  mutate(percent = count/value) |> 
  mutate(name = factor(name, levels = rev(orderOverview$name))) |> 
  ggbarplot(y = "percent", x = "continent", fill = "continent", alpha = "report", facet.by = "name", palette = palContinents, ncol = 1,
            rotate = T, order = names(palContinents), panel.labs = list(name = rev(orderOverview$label)),
            ylab = "% Reported", xlab = "") + 
  theme_pubclean(base_size = textSize/2) +
  theme(legend.title = element_blank()) -> p2

plot_grid(p1, p2, align = "hv", axis = "tblr")

```
 

 
### Overview of variant reporting
    
```{r reportVariants, fig.width=20, fig.height=10}
data.essentiality |> 
  filter(section == "Variants") |> 
  left_join(data.reported, by = c("group", "section", "name", "continent")) |> 
  mutate(label = ifelse((continent == "Worldwide" & normEssent >= cutoff), name, "")) |> 
  ggscatter(x = "normEssent", y = "percReported", palette = palContinents, 
            color = "continent", size = 4,
            xlab = "Score of Essentiality (SE)", ylab = "% Reported") + 
  lims(x = c(0, 1), y = c(0,1))+
  geom_abline(slope = 1, intercept = 0, lty = "dashed", size = 1, color = "grey23")+
  geom_label_repel(aes(label = label, color = continent), box.padding = 1, max.overlaps = Inf, 
                   show.legend = F, size = textSize/4)+
  theme_pubclean(base_size = textSize/2) + 
  theme(legend.position = "none") -> p1

data.reported |> 
  filter(section == "Variants") |> 
  pivot_longer(cols = responses.reported, values_to = "count", names_to = "report") |> 
  mutate(percent = count/value) |> 
  mutate(name = factor(name, levels = rev(orderVariants$name))) |> 
  ggbarplot(y = "percent", x = "continent", fill = "continent", alpha = "report", facet.by = "name", palette = palContinents, ncol = 1,
            rotate = T, order = names(palContinents), panel.labs = list(name = rev(orderVariants$label)),
            ylab = "% Reported", xlab = "") + 
  theme_pubclean(base_size = textSize/2) +
  theme(legend.title = element_blank()) -> p2

plot_grid(p1, p2, align = "hv", axis = "tblr")

```

Row
-------------------------------------    

### Overview of functional assessment
    
```{r reportFunctional, fig.width=20, fig.height=10}
data.essentiality |> 
  filter(section == "Functional") |> 
  left_join(data.reported, by = c("group", "section", "name", "continent")) |> 
  mutate(label = ifelse((continent == "Worldwide" & normEssent >= cutoff), name, "")) |> 
  ggscatter(x = "normEssent", y = "percReported", palette = palContinents, 
            color = "continent", size = 4,
            xlab = "Score of Essentiality (SE)", ylab = "% Reported") + 
  lims(x = c(-0.16, 1), y = c(0,1))+
  geom_abline(slope = 1, intercept = 0, lty = "dashed", size = 1, color = "grey23")+
  geom_label_repel(aes(label = label, color = continent), box.padding = 1, max.overlaps = Inf, 
                   show.legend = F, size = textSize/4)+
  theme_pubclean(base_size = textSize/2) + 
  theme(legend.position = "none") -> p1

data.reported |> 
  filter(section == "Functional") |> 
  pivot_longer(cols = responses.reported, values_to = "count", names_to = "report") |> 
  mutate(percent = count/value) |> 
  mutate(name = factor(name, levels = rev(orderFunctional$name))) |> 
  ggbarplot(y = "percent", x = "continent", fill = "continent", alpha = "report", facet.by = "name", palette = palContinents, ncol = 1,
            rotate = T, order = names(palContinents), panel.labs = list(name = rev(orderFunctional$label)),
            ylab = "% Reported", xlab = "") + 
  theme_pubclean(base_size = textSize/2) +
  theme(legend.title = element_blank()) -> p2

plot_grid(p1, p2, align = "hv", axis = "tblr")
```

### Treatment recommendations

    
```{r reportTreatment, fig.width=20, fig.height=10}
data.essentiality |> 
  filter(section == "Treatment") |> 
  left_join(data.reported, by = c("group", "section", "name", "continent")) |> 
  mutate(label = ifelse((continent == "Worldwide" & normEssent >= cutoff), name, "")) |> 
  ggscatter(x = "normEssent", y = "percReported", palette = palContinents, 
            color = "continent", size = 4,
            xlab = "Score of Essentiality (SE)", ylab = "% Reported") + 
  lims(x = c(0, 1), y = c(0,1))+
  geom_abline(slope = 1, intercept = 0, lty = "dashed", size = 1, color = "grey23")+
  geom_label_repel(aes(label = label, color = continent), box.padding = 1, max.overlaps = Inf, 
                   show.legend = F, size = textSize/6)+
  theme_pubclean(base_size = textSize/2) + 
  theme(legend.position = "none") -> p1

data.reported |> 
  filter(section == "Treatment") |> 
  pivot_longer(cols = responses.reported, values_to = "count", names_to = "report") |> 
  mutate(percent = count/value) |> 
  mutate(name = factor(name, levels = rev(orderTreatment$name))) |> 
  ggbarplot(y = "percent", x = "continent", fill = "continent", alpha = "report", facet.by = "name", palette = palContinents, ncol = 1,
            rotate = T, order = names(palContinents), panel.labs = list(name = rev(orderTreatment$label)),
            ylab = "% Reported", xlab = "") + 
  theme_pubclean(base_size = textSize/2) +
  theme(legend.title = element_blank()) -> p2

plot_grid(p1, p2, align = "hv", axis = "tblr")
```


Data
=====================================  
### Participants

```{r tableParticipants}
sum.country |> 
  arrange(desc(n)) |> 
  datatable(caption = "Overview of the number of survey responses for each country.")
```

### Essentiality
```{r tableEssentiality}
data.essentiality |> 
  datatable(caption = "Overview of the essentiality calculated by continent.")
```


Methods
===================================== 

Differences in country names were detected manually and harmonized to allow grouping.  Questions were organized into groups, based on topic of the presented elements. These groups include

- general information,
- variant information with a focus on SNVs and small indels,
- functional annotation,
- clinical information,
- aspects of clinical trials, and 
- more complex molecular profiles.

Each element of an NGS report can be defined as a) essential (E), b) preferred, but not essential (P) or c) not recommended (N). The responses are summarized as a score of essentiality (SE), which is normalized between the range of -1 (contra-productive) and 1 (essential): 

$$
SE = \frac{(E+ 0.5\cdot P − N)}{\text{#responses}}
$$

Email addresses were removed before uploading the final data set. 

In total `r nrDoubleResponders` participants filled out the survey twice over the course of the project. For the main analysis the most recent response has been selected. A comparison of consistency between is analysed separately.

A cutoff of `r cutoff` has been defined to identify an elementary NGS report. Based on the results coming from different regions, the elements can and should be adjusted.

